src/web/UserPresence.ts

```diff
@@ -18,21 +18,24 @@ if (typeof(document) !== 'undefined') {
 
 export class UserPresence extends RX.UserPresence {
     private _isPresent: boolean;
+    private _isAppFocused: boolean;
 
     constructor() {
         super();
         // Handle test environment where document is not defined.
         if (typeof(document) !== 'undefined') {
-            this._isPresent = ifvisible.now();
+            this._isPresent = this._isAppFocused = ifvisible.now();
 
             ifvisible.on('wakeup', this._handleWakeup.bind(this));
             ifvisible.on('idle', this._handleIdle.bind(this));
             ifvisible.on('focus', this._handleFocus.bind(this));
             ifvisible.on('blur', this._handleBlur.bind(this));
 
             window.addEventListener('blur', this._handleWindowBlur.bind(this));
+            window.addEventListener('focus', this._handleFocus.bind(this));
```

Comment:
Looking at the source for ifvisible, it looks like that module already installs a window-level "focus" and "blur" handler and uses them to report "focus" and "blur" events, which we subscribe to. So this appears to be redundant. See https://github.com/serkanyersen/ifvisible.js/blob/master/src/ifvisible.js. Ideally, we could simplify the existing logic by removing the _handleWindowBlur method and relying only on ifvisible to perform all of the work here.

Reply:
Hi, just wanted to point out that we're using https://www.npmjs.com/package/ifvisible (1.1.0), not https://www.npmjs.com/package/ifvisible.js - since this package was not updated and does not contain the fix for properly clearing timeouts.

So the actual source is here: https://github.com/muhammaddadu/ifvisible/blob/master/src/ifvisible.js

Also, IIRC ifvisible was not handling window.focus/window.blur as we needed, which is the reason we added our own. Cannot remember the details, but found this issue in the original repo: https://github.com/serkanyersen/ifvisible.js/issues/42

Reply:
@erictraut currently the only issue i see is not being able to detect that window is out of focus. It is also used in other places like to detect app activation state. 

Also it has logic to determine that user is inactive in the app - has a timeout for mouse moves. Which is valuable. 

Do you still think we need to reimplement this?

Reply:
I will reimplement ifvisibile here based on your comments in this PR. https://github.com/Microsoft/reactxp/pull/681

Reply:
Reimplemented ivisible in our own utils class AppVisibilityUtils. Please have a look.